segmentstore/storage/src/test/java/io/pravega/segmentstore/storage/chunklayer/SystemJournalTests.java

```diff
@@ -1191,6 +1196,99 @@ public void testSimpleTruncation() throws Exception {
     }
 
 
+    /**
+     * Test concurrent writes to storage system segments by simulating concurrent writes.
+     *
+     * @throws Exception Throws exception in case of any error.
+     */
+    @Test
+    public void testSystemSegmentConcurrency() throws Exception {
+        @Cleanup
+        ChunkStorage chunkStorage = getChunkStorage();
+        @Cleanup
+        ChunkMetadataStore metadataStoreBeforeCrash = getMetadataStore();
+        @Cleanup
+        ChunkMetadataStore metadataStoreAfterCrash = getMetadataStore();
+
+        int containerId = 42;
+        int maxLength = 8;
+        long epoch = 1;
+        val policy = new SegmentRollingPolicy(maxLength);
+        val config = getDefaultConfigBuilder(policy).build();
+
+        val snapshotData = new InMemorySnapshotInfoStore();
+        val snapshotInfoStore = new SnapshotInfoStore(containerId,
+                snapshotId -> snapshotData.setSnapshotId(containerId, snapshotId),
+                () -> snapshotData.getSnapshotId(containerId));
+
+        // Start container with epoch 1
+        @Cleanup
+        ChunkedSegmentStorage segmentStorage1 = new ChunkedSegmentStorage(containerId, chunkStorage, metadataStoreBeforeCrash, executorService(), config);
+
+        segmentStorage1.initialize(epoch);
+
+        // Bootstrap
+        segmentStorage1.bootstrap(snapshotInfoStore).join();
+        segmentStorage1.getGarbageCollector().deleteGarbage(false, 1000).get();
+
+        checkSystemSegmentsLayout(segmentStorage1);
+
+        // Simulate some writes to system segment, this should cause some new chunks being added.
+        // Simulate some writes to system segment, this should cause some new chunks being added.
+        val n = 4;
+        val data = new byte[n][100];
+
+        Random rnd = new Random();
```

Comment:
In unit tests, always initialize `new Random` with a constant - that will make your test predictable and reproducible. For example, if you get a test failure due to some specific input, using a Random without constant seed will guarantee that you will never see that input ever again, so reproducing a (sporadic) issue becomes much, much harder in that case.